server.py

```diff
@@ -15,64 +19,105 @@
 
 class Handler(BaseHTTPRequestHandler):
 
-    def _charge(self):
+    def _get_auth_cookies(self):
+        cookies = SimpleCookie(self.headers.get('Cookie'))
+        is_exist = False
+
+        try:
+            if int(cookies['auth'].value) > 0:
+                is_exist = True
+        except KeyError:
+            print('There is no entity with \'auth\' key')
+
+        return is_exist
+
+    def _set_template(self, page, **kwargs):
+        file_loader = FileSystemLoader(sep.join((path.dirname(path.abspath(__file__)), 'templates')))
+        env = Environment(loader=file_loader)
+        template = env.get_template(page)
+        output = template.render(kwargs)
+        self.wfile.write(output.encode('UTF-8'))
+
+    def _set_auth_cookies(self, value):
+        cookies = SimpleCookie()
+
+        cookies['auth'] = value
+
+        if value == 0:
+            cookies['auth']['expires'] = 'Thu, 01 Jan 1970 00:00:00 GMT'
+
+        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+
+    def _charge_handler(self):
         pass
 
-    def _error(self):
+    def _error_handler(self):
         pass
 
-    def _form(self):
+    def _form_handler(self):
         pass
 
-    def _log_in(self):
-        cookies = SimpleCookie()
-        cookies['auth'] = 100
-        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+    def _log_in_handler(self):
+        self._set_auth_cookies(100)
 
-    def _log_out(self):
-        cookies = SimpleCookie()
-        cookies['auth'] = 0
-        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+    def _log_out_handler(self):
+        self._set_auth_cookies(0)
 
-    PAGES = {
-        'charge.html': _charge,
-        'error.html': _error,
-        'form.html': _form,
-        'logIn.html': _log_in,
-        'logOut.html': _log_out,
+    HANDLERS = {
+        'error.html': _error_handler,
     }
 
-    def _set_headers(self, page=None, response_code=200, content_type='text/html'):
+    if PORT == 8001:
```

Comment:
It's not a requirement, i.e. you don't have to have specific urls on specific servers.
Just have all urls on all servers.
The real point is to organize communication between the parts of the solution as per the task requirements.